components/pages/Comment.js

```diff
@@ -1,19 +1,130 @@
+    paddingHorizontal: 15,
+    backgroundColor: '#fff',
+  },
+  input: {
+    flex: 1,
+    borderWidth: 1,
+    borderColor: '#ddd',
+    borderRadius: 5,
+    paddingHorizontal: 10,
+    paddingVertical: 5,
+    marginRight: 10,
+  },
+});
+
+export default function CommentsList({ route, navigation }) {
+  const { post } = route.params;
+  const [comments, setComments] = useState(route.params.comments);
+  const [likedComments, setLikedComments] = useState([]);
+  const [newComment, setNewComment] = useState('');
+
+  const commenter = useSelector((state) => state.user).data;
+
+  const handleCommentLike = async (comment) => {
+    const isLiked = likedComments.includes(comment.comment_id);
+    const updatedLifts = isLiked ? comment.lifts - 1 : comment.lifts + 1;
+    const updatedComments = comments.map(
+      (c) => (c.comment_id === comment.comment_id ? { ...c, lifts: updatedLifts } : c),
+    );
+
+    setComments(updatedComments);
+
+    if (isLiked) {
+      setLikedComments(likedComments.filter((id) => id !== comment.comment_id));
+    } else {
+      setLikedComments([...likedComments, comment.comment_id]);
+    }
+
+    try {
+      const documentRef = doc(db, 'posts', post.id.toString());
+      await updateDoc(documentRef, {
+        comments: updatedComments,
+      });
+    } catch (error) {
+      console.error('Error updating comment like count:', error);
+    }
+  };
+
+  const handleAddComment = async () => {
+    if (!newComment.trim()) return;
+
+    const newCommentObj = {
+      comment_id: uuid.v4(),
+      user: {
+        id: commenter.id,
+        name: `${commenter.name.first} ${commenter.name.last}`,
+        profilePhoto: commenter.profile_photo,
+        username: commenter.username,
+      },
+      comment: newComment,
+      lifts: 0,
+      report: false,
+      date: new Date().getTime(),
+    };
+
+    const updatedComments = [...comments, newCommentObj];
+    setComments(updatedComments);
+    setNewComment('');
+    try {
+      const documentRef = doc(db, 'posts', post.id.toString());
+      await updateDoc(documentRef, {
+        comments: updatedComments,
+      });
+    } catch (error) {
+      console.error('Error adding comment:', error);
+    }
+  };
 
   return (
-    <ScrollView>
-      <Text>Post: {post.content}</Text>
-      {/* Render comments here */}
-      {comments.map((comment) => (
-        <View key={comment.comment_id}>
-          <Text>{comment.user.username}: {comment.comment}</Text>
-        </View>
-      ))}
-      <TextInput placeholder="Add a comment" />
-    </ScrollView>
+    <SafeAreaView style={styles.container}>
+      <ScrollView>
+        <Post post={post} />
+        {comments.map((comment) => (
+          <CommentCard
+            key={comment.comment_id}
+            comment={comment}
+            onLike={() => handleCommentLike(comment)}
+            liked={likedComments.includes(comment.comment_id)}
+            navigation={navigation}
+          />
+        ))}
```

Comment:
might want to set a condition in case theres no comments (unless this whole component is already wrapped in that condition)